arm-mediaservices/2015-10-01/swagger/media.json

```diff
@@ -0,0 +1,715 @@
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "mediaServiceName",
+            "description": "Name of the Media Service.",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-z0-9]",
+            "minLength": 3,
+            "maxLength": 24
+          },
+          {
+            "name": "SyncStorageKeysInput",
+            "description": "Properties needed to sycnronize the keys for a storage account to the Media Service.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SyncStorageKeysInput"
+            }
+          }
+        ]
+      }
+    }
+  },
+  "definitions": {
+    "ApiEndpoint": {
+      "description": "The properties for a Media Services REST API endpoint.",
+      "properties": {
+        "endpoint": {
+          "description": "The Media Services REST endpoint.",
+          "type": "string"
+        },
+        "majorVersion": {
+          "description": "The version of Media Services REST API.",
+          "type": "string"
+        }
+      }
+    },
+    "ApiError": {
+    "description": "The error returned from a failed Media Services REST API call.",
+      "properties": {
+        "code": {
+          "description": "Error code.",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message",
+          "type": "string"
+        }
+      }
+    },
+    "CheckNameAvailabilityInput": {
+      "description": "The request body for CheckNameAvailability API.",
+      "properties": {
+        "name": {
+          "description": "Specifies the type of the resource.",
+          "type": "string"
+        },
+        "type": {
+          "description": "Specifies the type of the resource.",
+          "type": "string"
+        }
+      }
+    },
+    "CheckNameAvailabilityOutput": {
+      "description": "The request body for CheckNameAvailability API.",
+      "properties": {
+        "nameAvailable": {
+          "description": "Specifiies if the name is available.",
+          "type": "boolean"
+        },
+        "reason": {
+          "description": "Specifiies the reason if the name is not available",
+          "type": "string",
+          "enum": [
+            "None",
+            "Invalid",
+            "AlreadyExists"
+          ],
+          "x-ms-enum": {
+            "name": "EntityNameUnavailabilityReason",
+            "modelAsString": false
+          }  
+        },
+        "message": {
+          "description": "Specifiies the detailed reason if the name is not available",
+          "type": "string"
+        }
+      }
+    },
+    "MediaService": {
+      "description": "The properties of a Media Service resource.",
+      "properties": {
+        "properties": {

```

Comment:
It is recommended to use "x-ms-client-flatten" extension over here. This will bring the properties of MediaServiceProperties one level up during deserialization and down while serializing it, thus making it easier for customers.
Take a look at this [Storage example](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json#L760).
More info about this extension can be found [here](https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md#x-ms-client-flatten).